File: Backend/ProjectK.Backend/ProjectK.Common/Models/Dtos/MemberAwardDto.cs
using ProjectK.Common.Models.Enums;
using System;

namespace ProjectK.Common.Models.Dtos
{
    public class MemberAwardDto
    {
        public Guid MemberAwardKey { get; set; }
        public Guid MemberKey { get; set; }
        public Guid KurinKey { get; set; }
        public MemberAwardLevel Level { get; set; }
        public DateTime DateAcquired { get; set; }
        public string? Note { get; set; }
        public BadgeProgressStatus Status { get; set; }
        public string? ImageUrl { get; set; }
        public DateTime? SubmittedAtUtc { get; set; }
        public Guid? SubmittedByUserKey { get; set; }
        public DateTime? ReviewedAtUtc { get; set; }
        public Guid? ReviewedByUserKey { get; set; }
    }
}

File: Frontend/projectk-frontend/src/app/features/kurinModule/models/member-award.dto.ts
import { BadgeProgressStatus } from "./enums/badge-progress-status.enum";
import { MemberAwardLevel } from "./enums/member-award-level.enum";

export interface MemberAwardDto {
    memberAwardKey: string;
    memberKey: string;
    kurinKey: string;
    level: MemberAwardLevel;
    dateAcquired: string;
    note?: string;
    status: BadgeProgressStatus;
    imageUrl?: string;
    submittedAtUtc?: string;
    submittedByUserKey?: string;
    reviewedAtUtc?: string;
    reviewedByUserKey?: string;
}
